<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1135</width>
    <height>701</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>1091</width>
      <height>611</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Part1</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_openImg1">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>30</y>
        <width>161</width>
        <height>71</height>
       </rect>
      </property>
      <property name="text">
       <string>OpemImg</string>
      </property>
     </widget>
     <widget class="QLabel" name="label2">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>320</y>
        <width>320</width>
        <height>240</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_apply">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>220</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Apply</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>30</y>
        <width>320</width>
        <height>240</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Original image</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBox">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>130</y>
        <width>80</width>
        <height>22</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>RGB</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>CMY</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>HSI</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>XYZ</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>L*a*b</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>YUV</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label3">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>320</y>
        <width>320</width>
        <height>240</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label4">
      <property name="geometry">
       <rect>
        <x>710</x>
        <y>320</y>
        <width>320</width>
        <height>240</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>290</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Emoji</family>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::DefaultContextMenu</enum>
      </property>
      <property name="text">
       <string>Plane1</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>290</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Emoji</family>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::DefaultContextMenu</enum>
      </property>
      <property name="text">
       <string>Plane2</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>830</x>
        <y>290</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Emoji</family>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::DefaultContextMenu</enum>
      </property>
      <property name="text">
       <string>Plane3</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Part2 &amp; Part3</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_openImg2">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>30</y>
        <width>161</width>
        <height>71</height>
       </rect>
      </property>
      <property name="text">
       <string>OpemImg</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_colorbar">
      <property name="geometry">
       <rect>
        <x>910</x>
        <y>30</y>
        <width>20</width>
        <height>240</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label6">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>30</y>
        <width>320</width>
        <height>240</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Pseudo Color image</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_pseudo_color">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>180</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Pseudo Color</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>930</x>
        <y>30</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Emoji</family>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::DefaultContextMenu</enum>
      </property>
      <property name="text">
       <string>255</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>930</x>
        <y>250</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Emoji</family>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::DefaultContextMenu</enum>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>140</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="decimals">
       <number>3</number>
      </property>
      <property name="minimum">
       <double>1.047000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.047000000000000</double>
      </property>
      <property name="value">
       <double>1.047000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label5">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>30</y>
        <width>320</width>
        <height>240</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Original image</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>110</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Color Table adjustion</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_openImg3">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>330</y>
        <width>161</width>
        <height>71</height>
       </rect>
      </property>
      <property name="text">
       <string>OpemImg</string>
      </property>
     </widget>
     <widget class="QLabel" name="label7">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>310</y>
        <width>320</width>
        <height>240</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Original image</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label8">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>310</y>
        <width>320</width>
        <height>240</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Kmeans image</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_kmeans">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>500</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Kmeans</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_k">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>450</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="value">
       <number>2</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>420</y>
        <width>41</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>K</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_iter">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>450</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>10</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="singleStep">
       <number>10</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>420</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Max_iter</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>-10</y>
      <width>141</width>
      <height>111</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Adobe Heiti Std</family>
      <pointsize>26</pointsize>
     </font>
    </property>
    <property name="text">
     <string>HW5</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>50</y>
      <width>161</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Palace Script MT</family>
      <pointsize>26</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>TA so handsome</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1135</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
